internal/pipe/discourse/discourse.go

```diff
@@ -0,0 +1,112 @@
+// Package discourse announces to a Discourse forum.
+package discourse
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"net/http"
+
+	"github.com/caarlos0/env/v11"
+	"github.com/caarlos0/log"
+	"github.com/goreleaser/goreleaser/v2/internal/tmpl"
+	"github.com/goreleaser/goreleaser/v2/pkg/context"
+)
+
+const (
+	defaultTitleTemplate   = `{{ .ProjectName }} {{ .Tag }} is out!`
+	defaultMessageTemplate = `{{ .ProjectName }} {{ .Tag }} is out! Check it out at {{ .ReleaseURL }}`
+	defaultUsername        = "system"
+)
+
+type Pipe struct{}
+
+func (Pipe) String() string { return "discourse" }
+func (Pipe) Skip(ctx *context.Context) (bool, error) {
+	enable, err := tmpl.New(ctx).Bool(ctx.Config.Announce.Discourse.Enabled)
+	return !enable, err
+}
+
+type Config struct {
+	APIKey string `env:"DISCOURSE_API_KEY,notEmpty"`
+}
+
+func (Pipe) Default(ctx *context.Context) error {
+
+	if ctx.Config.Announce.Discourse.TitleTemplate == "" {
+		ctx.Config.Announce.Discourse.TitleTemplate = defaultTitleTemplate
+	}
+
+	if ctx.Config.Announce.Discourse.MessageTemplate == "" {
+		ctx.Config.Announce.Discourse.MessageTemplate = defaultMessageTemplate
+	}
+
+	if ctx.Config.Announce.Discourse.Username == "" {
+		ctx.Config.Announce.Discourse.Username = defaultUsername
+	}
+
+	return nil
+}
+
+func (p Pipe) Announce(ctx *context.Context) error {
+
+	title, err := tmpl.New(ctx).Apply(ctx.Config.Announce.Discourse.TitleTemplate)
+	if err != nil {
+		return fmt.Errorf("%s: %w", p, err)
```

Comment:
Other announcers handle the error message like this: `return fmt.Errorf("bluesky: %w", err)`. This has lead to an issue where the Mattermost announcer, which must have started as a copy from the Teams announcer, left this line in: https://github.com/goreleaser/goreleaser/blob/main/internal/pipe/mattermost/mattermost.go#L63

This will give an incorrect error message to users. My question is:

1. Can I leave my method that I used here?
2. Can I make a new PR changing the other announcers to use my method?
3. Or should I undo what I did here and make it work like the other announcers?

Reply:
> Can I make a new PR changing the other announcers to use my method?

yes this is probably the right choice